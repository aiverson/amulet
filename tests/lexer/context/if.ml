(* If expressions allow for multiple blocks *)
let _ =
  if true then
    1
    1
  else
    1
    1

(* Allow if/else on a new line *)
let _ =
  if true
  then
    1
  else
    1

(* Allow "early return" ifs *)
let _ =
  if true then 1 else
  if true then 1 else
    1
    1

(* Allow "else if" style expressions *)
let _ =
  if true then
    1
  else if true then
    1
  else
    1

(* Nesting *)
let _ =
  if true then
    if true then
      1
    else
      1
  else
    1

(*
(* TODO: Weird alignment of ifs *)
let _ = if true
  1
else
  2
*)
