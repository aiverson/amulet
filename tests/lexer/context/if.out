
let _ =
  if true then
    1
    $sep 1
  else
    1
    $sep 1


$end $end $sep let _ =
  if true
  then
    1
  else
    1


$end $end $sep let _ =
  if true then 1 else
  if true then 1 else
    1
    $sep 1


$end $end $end $sep let _ =
  if true then
    1
  else if true then
    1
  else
    1


$end $end $end $sep let _ =
  if true then
    if true then
      1
    else
      1
  $end else
    1








$end $end <eof>
